### ISSUE TITLE
[freerdp-shadow-cli] Does not work with Windows RDP Client or Guacamole

### ISSUE THREAD

**Issue by DaAwesomeP** [FreeRDP/FreeRDP#8744]
**Describe the bug**
I cannot keep connection open to freerdp-shadow-cli from Windows RDP Client or Guacamole but Remmina works.

Error shown:
```
[17:59:18:942] [327680:327680] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[17:59:18:944] [327680:327680] [INFO][com.freerdp.core.listener] - Listening on [0.0.0.0]:3389
[17:59:23:623] [327680:327869] [INFO][com.freerdp.core.connection] - Client Security: NLA:0 TLS:1 RDP:0
[17:59:23:623] [327680:327869] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[17:59:23:623] [327680:327869] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.connection] - Accepted client: Guacamole RDP
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.connection] - Accepted channels:
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.connection] -  cliprdr
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.connection] -  rdpdr
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.connection] -  rdpsnd
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
[17:59:23:678] [327680:327869] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
[17:59:24:650] [327680:327869] [INFO][com.freerdp.client.shadow] - Client from  is activated (1920x1080@16)
[17:59:25:369] [327680:327869] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[17:59:25:369] [327680:327869] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[17:59:25:369] [327680:327869] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
```
mstsc.exe and guacd seem to think that the server closed the connection.

**To Reproduce**
1. Install `freerdp2-shadow-x11` on Debian 11
2. `/usr/bin/freerdp-shadow-cli -auth`
3. Connect with Windows RDP Client or guacd.
4. You will see about 5 frames of active connection and then the connection will close.

**Expected behavior**
Connection should remain open.

**Screenshots**
N/A, see log above.

**Application details**
* FreeRDP version (`freerdp-shadow-cli /version`): `FreeRDP version 2.3.0 (git 2.3.0)`
* Command line used: `/usr/bin/freerdp-shadow-cli -auth`
* Output of `freerdp-shadow-cli /buildconfig`:
    ```
    [17:53:57:332] [315466:315466] [INFO][com.freerdp.server.shadow] - Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_GETLOGIN_R=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_URBDRC=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=TRUE WITH_FFMPEG=TRUE WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=ON WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=ON WITH_ICU=ON WITH_IPP=OFF WITH_JPEG=ON WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PAM=ON WITH_PCSC=ON WITH_PROFILER=OFF WITH_PROXY=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SHADOW=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=ON WITH_THIRD_PARTY=OFF WITH_VAAPI=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_WINPR_TOOLS=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XDAMAGE=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=ON WITH_XV=ON WITH_ZLIB=ON
    Build type:          RelWithDebInfo
    CFLAGS:              -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g -fno-omit-frame-pointer -DWINPR_DLL
    Compiler:            GNU, 10.2.1
    Target architecture: x64
    ```
* OS version connecting to (server side): Debian 11
* If available the log output from a run with `/log-level:trace 2>&1 | tee log.txt` : Doesn't work for `freerdp-shadow-cli`
* If you built it yourself add some notes which tag/commit/branch you have used, also your cmake parameters and
  compiler can help: N/A, installed via `freerdp2-shadow-x11` package

**Environment (please complete the following information):**
 - OS: Linux
 - Version/Distribution: Debian 11
 - Architecture: amd64

**Additional context**
N/A

**Comment by DaAwesomeP** [comment]
I think this is different than #7226 and #7232 because I am getting a fully functioning RDP connection (I can see the screen and a few frames of movement) but am getting disconnected after about 1 second.

**Comment by DaAwesomeP** [comment]
Doing some initial tracing purely through GitHub search:

1. The failed `peer_recv_pdu()` is this function, which leads me to believe that `tpkt_verify_header(s)` is returning an error
  https://github.com/FreeRDP/FreeRDP/blob/0445eec0bc9d528623e6e2d82ec039e9a4bfa5e7/libfreerdp/core/peer.c#L690-L700
2. The function `tpkt_verify_header()` returns an error when `Stream_CheckAndLogRequiredLength(TAG, s, 1)` returns false.
  https://github.com/FreeRDP/FreeRDP/blob/0445eec0bc9d528623e6e2d82ec039e9a4bfa5e7/libfreerdp/core/tpkt.c#L62-L83
3. This begins a string of macros:
  https://github.com/FreeRDP/FreeRDP/blob/075506f6c8d71393a792a69d104da6f38a47a5b2/winpr/include/winpr/stream.h#L94-L102


It seems that `Stream_CheckAndLogRequiredLengthOfSize()` is used all over the place, so I guess it is unlikely there is a significant bug in that function and more likely that these clients are sending something unexpected?

What is the best way to proceed here? I hope that someone can reproduce this, but I can look into trying to get a WireShark capture. Maybe if I set the resolution very low it won't be a massive capture.

**Comment by akallabeth** [comment]
@DaAwesomeP `freerdp-shadow` has some known issues, so better try with a more recent version.

**Comment by DaAwesomeP** [comment]
@akallabeth does the nightly CI Deb package build include it?

**Comment by akallabeth** [comment]
yes, but that is from `master` branch

**Comment by DaAwesomeP** [comment]
OK, tried using this build: http://pub.freerdp.com/repositories/deb/bullseye/pool/main/f/freerdp-nightly/freerdp-nightly_3.0.0%2B0~20230306024823.1284~1.gbpfb6014_amd64.deb

It seems the packaging CI for the non-nightly is not really run anymore? `http://pub.freerdp.com/repositories/deb/bullseye/ freerdp2 main` is stuck at v2.0.0.

This build seems to have an entirely separate issue with NLA not disabling.

With this build using `-auth` (or omitting it) causes Remmina to prompt for credentials when it should not:
```
[09:44:08:249] [200675:200679] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() IPPROTO_TCP, TCP_KEEPIDLE
[09:44:08:249] [200675:200679] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_KEEPCNT
[09:44:08:249] [200675:200679] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_KEEPINTVL
[09:44:08:249] [200675:200679] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_USER_TIMEOUT
[09:44:08:251] [200675:200679] [DEBUG][com.winpr.thread] - [ThreadCloseHandle]: Thread running, setting to detached state!
[09:44:08:251] [200675:200765] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_INITIAL
[09:44:08:254] [200675:200765] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[09:44:08:254] [200675:200765] [DEBUG][com.freerdp.core.nego] - [nego_process_negotiation_request]: RDP_NEG_REQ: RequestedProtocol: 0x00000003
[09:44:08:254] [200675:200765] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Client Security: NLA:1 TLS:1 RDP:0
[09:44:08:254] [200675:200765] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Server Security: NLA:1 TLS:1 RDP:1
[09:44:08:254] [200675:200765] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Negotiated Security: NLA:1 TLS:0 RDP:0
[09:44:08:323] [200675:200765] [DEBUG][com.winpr.sspi] - [InitSecurityInterfaceExA]: InitSecurityInterfaceExA
[09:44:08:323] [200675:200765] [DEBUG][com.freerdp.core.auth] - [credssp_auth_init]: Using package: Negotiate (cbMaxToken: 12256 bytes)
[09:44:08:324] [200675:200765] [DEBUG][com.freerdp.core.auth] - [credssp_auth_setup_server]: Acquired server credentials
[09:44:08:324] [200675:200765] [DEBUG][com.freerdp.core.nla] - [nla_set_state]: -- NLA_STATE_INITIAL    --> NLA_STATE_INITIAL
```

 Adding `-auth -sec-nla` causes:
```
[09:45:16:305] [203284:203288] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() IPPROTO_TCP, TCP_KEEPIDLE
[09:45:16:305] [203284:203288] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_KEEPCNT
[09:45:16:305] [203284:203288] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_KEEPINTVL
[09:45:16:306] [203284:203288] [WARN][com.freerdp.core] - [freerdp_tcp_set_keep_alive_mode]: setsockopt() SOL_TCP, TCP_USER_TIMEOUT
[09:45:16:307] [203284:203288] [DEBUG][com.winpr.thread] - [ThreadCloseHandle]: Thread running, setting to detached state!
[09:45:16:307] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_INITIAL
[09:45:16:311] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[09:45:16:311] [203284:203350] [DEBUG][com.freerdp.core.nego] - [nego_process_negotiation_request]: RDP_NEG_REQ: RequestedProtocol: 0x00000003
[09:45:16:311] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Client Security: NLA:1 TLS:1 RDP:0
[09:45:16:311] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Server Security: NLA:0 TLS:1 RDP:1
[09:45:16:311] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_nego]: Negotiated Security: NLA:0 TLS:1 RDP:0
[09:45:16:389] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_MCS_CREATE_REQUEST
[09:45:16:440] [203284:203350] [DEBUG][com.freerdp.core.gcc] - [gcc_read_client_core_data]: Received EarlyCapabilityFlags=RNS_UD_CS_SUPPORT_ERRINFO_PDU|RNS_UD_CS_WANT_32BPP_SESSION|RNS_UD_CS_VALID_CONNECTION_TYPE|RNS_UD_CS_SUPPORT_NETCHAR_AUTODETECT|RNS_UD_CS_SUPPORT_DYNVC_GFX_PROTOCOL|RNS_UD_CS_SUPPORT_HEARTBEAT_PDU|[0x000005a3]
[09:45:16:440] [203284:203350] [WARN][com.freerdp.common] - [freerdp_settings_set_pointer_len]: FreeRDP_ChannelDefArray::len expected to be >= 31, but have 4
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]: Accepted client: REDACTED
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]: Accepted channels:
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]:  rdpdr [1004]
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]:  rdpsnd [1005]
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]:  cliprdr [1006]
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.connection] - [rdp_server_accept_mcs_connect_initial]:  drdynvc [1007]
[09:45:16:440] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CREATE_REQUEST --> CONNECTION_STATE_MCS_CREATE_RESPONSE
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.gcc] - [gcc_write_server_security_data]: Active rdp encryption level: NONE
[09:45:16:440] [203284:203350] [INFO][com.freerdp.core.gcc] - [gcc_write_server_security_data]: Selected rdp encryption method: NONE
[09:45:16:440] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CREATE_RESPONSE --> CONNECTION_STATE_MCS_ERECT_DOMAIN
[09:45:17:534] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_ERECT_DOMAIN --> CONNECTION_STATE_MCS_ATTACH_USER
[09:45:17:534] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_ATTACH_USER --> CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM
[09:45:17:534] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:640] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:738] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:842] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:950] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:951] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:951] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:951] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:951] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:951] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:041] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:139] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpdr [1004]
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  rdpsnd [1005]
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  cliprdr [1006]
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_accept_mcs_channel_join_request]:  drdynvc [1007]
[09:45:17:237] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_RDP_SECURITY_COMMENCEMENT
[09:45:17:339] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_RDP_SECURITY_COMMENCEMENT --> CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE
[09:45:17:339] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE --> CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST
[09:45:17:339] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST --> CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_RESPONSE
[09:45:17:442] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_RESPONSE --> CONNECTION_STATE_LICENSING
[09:45:17:443] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_LICENSING --> CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST
[09:45:17:443] [203284:203350] [DEBUG][com.freerdp.core.connection] - [rdp_server_transition_to_state]: CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST --> CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_RESPONSE
[09:45:25:468] [203284:203350] [ERROR][com.freerdp.core.transport] - [transport_read_layer]: BIO_read returned a system error 0: Success
[09:45:25:469] [203284:203350] [ERROR][com.freerdp.core] - [freerdp_set_last_error_ex]: transport_read_layer ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[09:45:25:469] [203284:203350] [DEBUG][com.freerdp.core.transport] - [transport_check_fds]: transport_check_fds: transport_read_pdu() - -1
[09:45:25:469] [203284:203350] [DEBUG][com.freerdp.core.rdp] - [rdp_check_fds]: transport_check_fds() - -1
[09:45:25:469] [203284:203350] [ERROR][com.freerdp.client.shadow] - [shadow_client_thread]: Failed to check FreeRDP file descriptor
```

**Comment by akallabeth** [comment]
@DaAwesomeP you had bad timing, there was #8752 merged yesterday, which fixed a bug with `gfx` also sending frames as `surfacebits` (meaning `mstsc` just disconnected)

**Comment by akallabeth** [comment]
@DaAwesomeP also, if you want to use `NLA` you need `/sam-file:<somefile>` where you put in values like `winpr-hash -u <username> -p <password> [-d <domain>] -f sam -v 1`

**Comment by akallabeth** [comment]
working with current master 7e44d447f02ffd45b60b94a55d4fd90018842ec9

**Comment by DaAwesomeP** [comment]
@akallabeth Yup, that seems to work with Guacamole now! Thanks!

Probably a different issue (and not at all a showstopper), but Remmina is now prompting me for credentials even when using `-auth` unless I also use `-sec-nla`.

**Comment by akallabeth** [comment]
@DaAwesomeP we´ve fixed #8787 and #8782 so the client should ask for credentials now if it has none. (that has actually nothing to do with `-auth` but the fact that the client must ask for credentials before connection and we had a bug there
